keras_cv/models/object_detection/yolo_v8/yolo_v8.py

```diff
@@ -0,0 +1,604 @@
+    out = conv_bn(
+        out,
+        input_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_output",
+    )
+    return out
+
+
+# This should probably just be a CSPDarknet. That's an outstanding TODO
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class YOLOV8Backbone(Backbone):
+    def __init__(
+        self,
+        include_rescaling,
+        channels=[128, 256, 512, 1024],
+        depths=[3, 6, 6, 3],
+        input_shape=(512, 512, 3),
+        activation="swish",
+        **kwargs,
+    ):
+        inputs = layers.Input(input_shape)
+
+        x = inputs
+        if include_rescaling:
+            x = layers.Rescaling(1 / 255.0)(x)
+
+        """ Stem """
+        # stem_width = stem_width if stem_width > 0 else channels[0]
+        stem_width = channels[0]
+        nn = conv_bn(
+            x,
+            stem_width // 2,
+            kernel_size=3,
+            strides=2,
+            activation=activation,
+            name="stem_1",
+        )
+        nn = conv_bn(
+            nn,
+            stem_width,
+            kernel_size=3,
+            strides=2,
+            activation=activation,
+            name="stem_2",
+        )
+
+        """ blocks """
+        pyramid_level_inputs = {0: nn.node.layer.name}
+        features = {0: nn}
+        for stack_id, (channel, depth) in enumerate(zip(channels, depths)):
+            stack_name = f"stack{stack_id + 1}"
+            if stack_id >= 1:
+                nn = conv_bn(
+                    nn,
+                    channel,
+                    kernel_size=3,
+                    strides=2,
+                    activation=activation,
+                    name=f"{stack_name}_downsample",
+                )
+            nn = csp_with_2_conv(
+                nn,
+                depth=depth,
+                expansion=0.5,
+                activation=activation,
+                name=f"{stack_name}_c2f",
+            )
+
+            if stack_id == len(depths) - 1:
+                nn = spatial_pyramid_pooling_fast(
+                    nn,
+                    pool_size=5,
+                    activation=activation,
+                    name=f"{stack_name}_spp_fast",
+                )
+            pyramid_level_inputs[stack_id + 1] = nn.node.layer.name
+            features[stack_id + 1] = nn
+
+        super().__init__(inputs=inputs, outputs=features, **kwargs)
+        self.pyramid_level_inputs = pyramid_level_inputs
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return copy.deepcopy(yolo_v8_backbone_presets)
+
+    @classproperty
+    def presets_with_weights(cls):
+        """Dictionary of preset names and configurations that include weights."""
+        return copy.deepcopy(yolo_v8_backbone_presets_with_weights)
+
+
+def path_aggregation_fpn(features, depth=3, name=None):
```

Comment:
I thought YOLO only uses the last layer? Or are all YOLOs not alike?

Reply:
YOLOv8 does use multiple outputs from the backbone in an FPN, yes.

I think maybe some of the YOLO models don't? Probably @LukeWood or @leondgarse would know for sure

Reply:
> I thought YOLO only uses the last layer? Or are all YOLOs not alike?

I believe the latter is correct - that not all YOLOs are alike.  I believe only the first yolo (maybe first 2-3) only used the final feature map.

Reply:
Are these YOLOs still less computationally intensive / mobile-friendly than other models in that case? Or do they have as much in common as King Charles II and Charles III?